### PR TITLE
Merge from main-3 to main

### PR DESCRIPTION
**Pull request by corneil**

### CODE REVIEW

File: .github/workflows/build-snapshot-worker.yml [hunk 1]
```diff
@@ -38,14 +34,14 @@ jobs:
...
-        jfrog rt mvnc \
+        jfrog rt mvnc --use-wrapper \
           --server-id-resolve=repo.spring.io \
           --server-id-deploy=repo.spring.io \
           --repo-resolve-releases=libs-milestone \
           --repo-resolve-snapshots=libs-snapshot \
           --repo-deploy-releases=libs-release-local \
           --repo-deploy-snapshots=libs-snapshot-local
-        echo JFROG_CLI_BUILD_NAME=spring-cloud-deployer-main >> $GITHUB_ENV
+        echo JFROG_CLI_BUILD_NAME=spring-cloud-deployer-main-3 >> $GITHUB_ENV
```
Comment (ilayaperumalg) [inline .github/workflows/build-snapshot-worker.yml:44]: main ?

File: pom.xml [hunk 2]
```diff
@@ -52,22 +52,19 @@
...
-		<spring-boot.version>2.7.18</spring-boot.version>
-		<spring-framework.version>5.3.39</spring-framework.version>
+		<java.version>17</java.version>
+		<!-- @TODO boot3 waiting for 4.x to support jakarta ee -->
+		<maven.version>3.9.6</maven.version>
+		<!-- @TODO boot3 waiting for 4.x to support jakarta ee -->
+		<maven-resolver.version>1.9.18</maven-resolver.version>
+		<maven-wagon.version>3.5.3</maven-wagon.version>
+		<commons-io.version>2.16.1</commons-io.version>
+		<kubernetes-fabric8-client.version>6.9.2</kubernetes-fabric8-client.version>
```
Comment (ilayaperumalg) [inline pom.xml:62]: Is this fabric8 client version something we already tested against?

Comment (corneil) [reply pom.xml:62]: I did run main-3 ITs when I did the initial work. However now docker rate limiting is so back that we need to update the ITs to provide imagePullSecrets everywhere.

**Comment by onobc** [comment]
Closing in favor of https://github.com/spring-cloud/spring-cloud-deployer/pull/482

**Pull request closed without merging**